soda_machine.rb

```diff
@@ -1,18 +1,26 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_reader  :cash, :sodas, :sold_soda
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = nil
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    if @sodas.include? soda_brand
```

Comment:
I highly suggest familiarizing yourself with what data types you are working with before trying to manipulate them with your code. Here, `@sodas` is an `Array` of `Soda` objects, but `soda_brand` is a `String`. As such, you won't get any matches via `.include?`. You'll need to first access the `brand` attribute of the `Soda` object and compare that to `soda_brand`